### PR TITLE
Update ssh2 package & Ignore disconnection failures

### PR DESCRIPTION
**Pull request by lkillgore**
When we connect to an SSH server that's Windows, it will commonly error on the disconnect.  Elsewhere, the maintainer of the node SSH module we're using suggests to ignore the failures on logout.  Seems reasonable to me if we've successfully uploaded all of the files.

### CODE REVIEW

**Comment by lkillgore** [comment]
> Update the task versions too

Good catch!  Thanks!

File: Tasks/CopyFilesOverSSHV0/sshhelper.ts [hunk 1]
```diff
@@ -28,19 +28,19 @@ export class SshHelper {
         this.sshConfig = sshConfig;
     }
 
-    private setupSshClientConnection() : Q.Promise<void> {
-        var defer = Q.defer<void>();
+    private async setupSshClientConnection() : Promise<void> {
```
Comment (brcrista) [inline Tasks/CopyFilesOverSSHV0/sshhelper.ts]: +1

File: Tasks/CopyFilesOverSSHV0/sshhelper.ts [hunk 2]
```diff
@@ -28,19 +28,19 @@ export class SshHelper {
...
+        let defer = Q.defer<void>();
```
Comment (brcrista) [inline Tasks/CopyFilesOverSSHV0/sshhelper.ts]: Can be `const`?

File: Tasks/CopyFilesOverSSHV0/sshhelper.ts [hunk 3]
```diff
@@ -73,6 +73,9 @@ export class SshHelper {
     closeConnection() {
         try {
             if (this.sftpClient) {
+                this.sftpClient.on('error', (err) => {
+                    tl.debug('sftpClient: Ignoring error diconnecting: ' + JSON.stringify(err));
```
Comment (brcrista) [inline Tasks/CopyFilesOverSSHV0/sshhelper.ts]: Are you sure we want to serialize as JSON?  I think we just want `+ err` so it will `.toString()` it.

**Comment by brcrista** [comment]
I've been updating the tsconfig.json with the `"strict": true` family of flags as we work on tasks -- see #9378 for example.  We should consider enabling it here to catch more issues.

Note that task-lib isn't compiled with `strictNullChecks`, though, so it will get type inference wrong with that flag.

File: Tasks/CopyFilesOverSSHV0/sshhelper.ts (hunk 2 above)
Comment (lkillgore) [reply Tasks/CopyFilesOverSSHV0/sshhelper.ts]: 👍

**Pull request merged**